hyperspy/_signals/eds.py

```diff
@@ -222,16 +224,107 @@ def rebin(self, new_shape):
                    np.array(new_shape_in_array))
         s = super(EDSSpectrum, self).rebin(new_shape)
         # modify time per spectrum
+        if "Acquisition_instrument.SEM.Detector.EDS.real_time" in s.metadata:
+            for i, t in enumerate(s.axes_manager.navigation_axes):
+                s.metadata.Acquisition_instrument.SEM.Detector.EDS.real_time\
+                    *= factors[i]
         if "Acquisition_instrument.SEM.Detector.EDS.live_time" in s.metadata:
-            for factor in factors:
+            for i, t in enumerate(s.axes_manager.navigation_axes):
                 s.metadata.Acquisition_instrument.SEM.Detector.EDS.live_time\
-                    *= factor
+                    *= factors[i]
+        if "Acquisition_instrument.TEM.Detector.EDS.real_time" in s.metadata:
+            for i, t in enumerate(s.axes_manager.navigation_axes):
+                s.metadata.Acquisition_instrument.TEM.Detector.EDS.real_time\
+                    *= factors[i]
         if "Acquisition_instrument.TEM.Detector.EDS.live_time" in s.metadata:
-            for factor in factors:
+            for i, t in enumerate(s.axes_manager.navigation_axes):
                 s.metadata.Acquisition_instrument.TEM.Detector.EDS.live_time\
-                    *= factor
+                    *= factors[i]
         return s
 
+    def linear_bin(self, scale, crop='on'):
```

Comment:
I think it is in the BaseSignal. I've added this function to EDS, base and EELS. The EDS and EELS functions  specifically adjust the dwell time as well to account for the new bin sizes.